eoc_journal/eoc_journal.py

```diff
@@ -97,6 +110,44 @@ def student_view(self, context=None):
         )
         return fragment
 
+    def list_user_pb_answers_by_section(self, user_id, course_id):
+        """
+        Returns a list of dicts with pb-answers grouped by section.
+        """
+        # Get the selected blocks and their answers
+        blocks = self.list_pb_answers()
+        answers_names = [block['name'] for block in blocks]
+        course_key = CourseKey.from_string(course_id)
+        answers = Answer.objects.filter(
+            course_key=course_key,
+            student_id=user_id,
+            name__in=answers_names
+        )
+
+        if answers.count() > 0:
+            # Map answer names to student inputs
+            students_inputs = {a.name: a.student_input for a in answers}
+
+            # Group answers by section
+            answers = defaultdict(list)
+
+            for block in blocks:
+                name = block['name']
+                section = block['section']
+
+                answers[section].append({
+                    'answer': students_inputs[name],
```

Comment:
This will fail if student did not answer the question yet. We should check if `name in student_inputs` before appending.